pkg/deployment/context_impl.go

```diff
@@ -63,20 +65,37 @@ func (d *Deployment) GetNamespace() string {
 	return d.apiObject.GetNamespace()
 }
 
+// GetPhase returns the current phase of the deployment
+func (d *Deployment) GetPhase() api.DeploymentPhase {
+	return d.status.last.Phase
+}
+
 // GetSpec returns the current specification
 func (d *Deployment) GetSpec() api.DeploymentSpec {
 	return d.apiObject.Spec
 }
 
 // GetStatus returns the current status of the deployment
-func (d *Deployment) GetStatus() api.DeploymentStatus {
-	return d.status
+// together with the current version of that status.
+func (d *Deployment) GetStatus() (api.DeploymentStatus, int32) {
+	version := atomic.LoadInt32(&d.status.version)
+	return *d.status.last.DeepCopy(), version
```

Comment:
What if `*d.status.last` (and indeed `*d.status.version`) have changed between the execution of line 81 and line 82? Do we not have to check the version again after the DeepCopy?
Or we have to make getting the version and doing a DeepCopy an atomic operation (with a Mutex)...